pages/_app.js

```diff
@@ -1,10 +1,42 @@
 import GlobalStyle from "../styles";
-
+import { useState } from "react";
+import { recipes } from "@/lib/recipes";
 export default function App({ Component, pageProps }) {
+  const [bookmarkedRecipesIDs, setBookmarkedRecipesIDs] = useState([]);
+
+  function checkIfRecipeIsBookmarked(id) {
+    return bookmarkedRecipesIDs.includes(id);
+  }
+
+  function addRecipeToBookmarked(id) {
+    setBookmarkedRecipesIDs([id, ...bookmarkedRecipesIDs]);
+  }
+
+  function removeRecipeFromBookmarked(id) {
+    setBookmarkedRecipesIDs(() => {
+      const RecipeIsBookmarkedWithoutCertainID = bookmarkedRecipesIDs.filter(
+        (item) => item !== id
+      );
+      return RecipeIsBookmarkedWithoutCertainID;
+    });
```

Comment:
if you don't want to use the "prev" value inside the callback function, you should refactor this code to reduce the nesting:

```js
  const updatedBookmarkedRecipesIDs = bookmarkedRecipesIDs.filter(
    (item) => item !== id
  );
  setBookmarkedRecipesIDs(updatedBookmarkedRecipesIDs);

```

Reply:
please refactor this code.